### PR TITLE
Add C implementation for `@stdlib/math/base/special/expit`

### LINKED ISSUE #719: [RFC]: Add C implementation for `@stdlib/math/base/special/erfc`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/erfc`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/erfc).

```c
double stdlib_base_erfc( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### LINKED ISSUE #767: [RFC]: Add C implementation for `@stdlib/math/base/special/expit`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/expit`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/expit).

```c
double stdlib_base_expit( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #767.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a review.

File: lib/node_modules/@stdlib/math/base/special/expit/README.md [hunk 1]
```diff
@@ -95,6 +95,96 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = (double) rand() / (double) RAND_MAX;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/README.md]:
```suggestion
        x = (double)rand() / (double)RAND_MAX;
```

File: lib/node_modules/@stdlib/math/base/special/expit/benchmark/c/native/benchmark.c [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+/**
+* Evaluates the expit function.
+*
+* @param x   input value
+* @return result
+*/
+double expit( double x ) {
+	return 1.0 / ( 1.0 + exp( -x ) );
+}
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/benchmark/c/native/benchmark.c]:
```suggestion
```

Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/benchmark/c/native/benchmark.c]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/expit/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = (double) rand() / (double) RAND_MAX;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expit/examples/c/example.c]:
```suggestion
        x = (double)rand() / (double)RAND_MAX;
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**